### PR TITLE
Button to clear the recent objects list

### LINKED ISSUE #6240: Clear the Recent Objects list
It would be nice if there were a way to clear Recent Items other than clearing the localStorage cache manually. This could be a button, context action, or other, that can either partially or entirely clear the Recent Objects list.

### PR DESCRIPTION
**Pull request by 360macky**
Closes #6240.

### Describe your changes:
- Adds a "Clear Recent Objects" button that removes the Recent Objects list from the Local Storage, and updates the panel to reflect the change in the user interface.
- Adds a functional test to check this behavior.

The button has an icon of "icon-clear-data".

https://user-images.githubusercontent.com/47821093/218269054-5ba07e05-9269-4705-ae71-20a2cb10da16.mov

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/ui/layout/Layout.vue
Comment (ozyx) [inline src/ui/layout/Layout.vue:291]: Let's also just call this `handleClearRecentObjects()`

**Review by ozyx — CHANGES_REQUESTED** [review]
Thanks so much, this is awesome! I've left just a few suggestions for improvements. We will also need to get @charlesh88's review on the visual aspect of the clear button, but should be good to go after that!

File: e2e/tests/functional/recentObjects.e2e.spec.js
Comment (360macky) [reply e2e/tests/functional/recentObjects.e2e.spec.js:202]: Brilliant idea!

**Comment by 360macky** [comment]
Thanks for the reviewing @ozyx. PR updated!

File: src/ui/layout/RecentObjectsList.vue
Comment (ozyx) [inline src/ui/layout/RecentObjectsList.vue:199]: Ah, sorry-- I think what I wrote in the first review was kind of confusing. 

We want to clear RecentObjects both in `localStorage` _and_ locally here. So let's just add the line back in:

```js
localStorage.removeItem(LOCAL_STORAGE_KEY__RECENT_OBJECTS);
this.recents = [];
```

File: e2e/tests/functional/recentObjects.e2e.spec.js
Comment (ozyx) [inline e2e/tests/functional/recentObjects.e2e.spec.js:194]: Nice!

**Review by ozyx — CHANGES_REQUESTED** [review]
Great! Just one final change request, and then it's ready for UI/UX review.

File: src/ui/layout/RecentObjectsList.vue
Comment (360macky) [reply src/ui/layout/RecentObjectsList.vue:199]: Ohh. Sorry, I thought we had changed strategy in this behavior. 😅
I put it back. 👍

**Review by charlesh88 — CHANGES_REQUESTED** [review]
This looks great, one change requested: let's just remove the `l-shell__reset-tree-button` class name from the button - this was name was originally applied to the reset tree button, but there's not actually a class for this, and in this context, it's actually the wrong name.

**Comment by 360macky** [comment]
Thank you @charlesh88. Done!

**Comment by charlesh88** [comment]
On further reflection, we should really invoke a confirmation dialog on this action - otherwise, the user can accidentally blow away all their recents if they click unintentionally. Let's please add a confirmation dialog that reads like this, with Ok and Cancel buttons:

**This action will clear the Recently View items list. Are you sure you want to continue?** 

You can look at the Clear Data button that's available in the status bar area at the top of the app for an example of something similar. 

Thanks again for this!

**Comment by 360macky** [comment]
Done @charlesh88! I just updated the behavior of the clear recent objects button with the dialog confirmation, as well as the test for this button.

Screenshot of the new dialog confirmation

File: src/ui/layout/RecentObjectsList.vue
Comment (ozyx) [inline src/ui/layout/RecentObjectsList.vue:203]:
```suggestion
                message: 'This action will clear the Recently Viewed Objects list. Are you sure you want to continue?',
```

File: src/ui/layout/RecentObjectsList.vue
Comment (ozyx) [inline src/ui/layout/RecentObjectsList.vue:210]:
```suggestion
                            localStorage.removeItem(LOCAL_STORAGE_KEY__RECENT_OBJECTS);
                            this.recents = [];
                            dialog.dismiss();
```

File: src/ui/layout/RecentObjectsList.vue
Comment (ozyx) [inline src/ui/layout/RecentObjectsList.vue:200]:
```suggestion
            const dialog = this.openmct.overlays.dialog({
```

**Review by ozyx — CHANGES_REQUESTED** [review]
OK! Sorry for the delay. I just have a few small suggestions before we merge.

**Comment by 360macky** [comment]
Thanks @ozyx! Done.

**Comment by ozyx** [comment]
> Thanks @ozyx! Done.

Awesome! We will be merging this in as soon as we cut the release branch for 2.2.0. Thanks again!

**Review by ozyx — APPROVED** [review]

**Review by charlesh88 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
